src/cargo/core/compiler/fingerprint/dirty_reason.rs

```diff
@@ -4,7 +4,16 @@ use std::fmt::Debug;
 use super::*;
 use crate::core::Shell;
 
-pub type DirtyReason = DirtyDetail;
+/// Tell why a unit needs to rebuild.
+#[derive(Clone, Debug)]
+pub enum DirtyReason {
+    /// First time to build something.
+    FreshBuild,
```

Comment:
FYI this use of "fresh" runs counter to the use at https://doc.rust-lang.org/nightly/nightly-rustc/cargo/core/compiler/enum.Freshness.html#variant.Fresh (yes, I prefer this ones use of "fresh")

Reply:
Oh, its existing. Great.